services/dns/record.go

```diff
@@ -0,0 +1,209 @@
+package dns
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/gofrs/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	dns "github.com/ionos-cloud/sdk-go-dns"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/utils"
+)
+
+var recordResourceName = "DNS Record"
+
+func (c *Client) CreateRecord(ctx context.Context, zoneId string, d *schema.ResourceData) (recordResponse dns.RecordResponse, responseInfo utils.ApiResponseInfo, err error) {
+	// Must is a helper that wraps a call to a function returning (UUID, error) and panics if the error is non-nil.
+	uuidV4 := uuid.Must(uuid.NewV4())
+	recordUuid := uuid.NewV5(uuid.NewV5(uuid.NamespaceURL, "https://github.com/ionos-cloud/terraform-provider-ionoscloud"), uuidV4.String())
```

Comment:
@cristiGuranIonos I had in mind to generate the record `UUID` also based on zone `UUID`, not sure if this adds anything useful tho, what do you think?

Reply:
I don't think it's necessary